applications/client/modules/procedureapi.py

```diff
@@ -12,22 +14,21 @@ def write_value(self, dictionary):
         """
         db = current.db
         for key,val in dictionary.iteritems():
-            val = dictionary[key]
             # Update the key value for this module if it already exists
             db.module_values.update_or_insert((db.module_values.name == key) & (db.module_values.modulename == self.procedure_name),
                                               time_stamp=datetime.datetime.utcnow(),
                                               modulename=self.procedure_name,
                                               name=key,
-                                              module_value=val)
+                                              module_value=json_plus.Serializable.dumps(val))
 
 
     def write_output(self, name, data, tag):
         """ This write the values and the tag to the outputs table.
         param name : This is the name of the output
-         param data : The value of the output
+         param data : The value of the output : This is a dictionary of key value pairs
```

Comment:
Why?  Let's discuss this.  I thought that once you have the name, you just specify a value, not a dictionary.